core/core-configuration-layer.el

```diff
@@ -1621,10 +1684,18 @@ wether the declared layer is an used one or not."
 LAYER must not be the owner of PKG."
   (let* ((owner (configuration-layer/get-layer (car (oref pkg :owners))))
          (disabled (oref owner :disabled-for))
-         (enabled (oref owner :enabled-for)))
-    (if (not (eq 'unspecified enabled))
-        (memq layer enabled)
-      (not (memq layer disabled)))))
+         (enabled (oref owner :enabled-for))
+         (depends (oref pkg :depends))
+         (depends (if (listp depends) depends (list depends))))
```

Comment:
Maybe we can do this data structure transformation when we store the slot info ?